x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pytorch/process/PyTorchResultProcessor.java

```diff
@@ -85,23 +90,30 @@ public void process(NativePyTorchProcess process) {
         } finally {
             pendingResults.forEach(
                 (id, pendingResult) -> pendingResult.listener.onResponse(
-                    new PyTorchResult(id, null, null, "inference canceled as process is stopping")
+                    new PyTorchInferenceResult(id, null, null, "inference canceled as process is stopping")
                 )
             );
             pendingResults.clear();
         }
         logger.debug(() -> new ParameterizedMessage("[{}] Results processing finished", deploymentId));
     }
 
-    public synchronized LongSummaryStatistics getTimingStats() {
-        return new LongSummaryStatistics(timingStats.getCount(), timingStats.getMin(), timingStats.getMax(), timingStats.getSum());
-    }
-
-    private synchronized void processResult(PyTorchResult result) {
-        if (result.isError() == false) {
-            timingStats.accept(result.getTimeMs());
+    private void processInferenceResult(PyTorchInferenceResult inferenceResult) {
+        logger.trace(() -> new ParameterizedMessage("[{}] Parsed result with id [{}]", deploymentId, inferenceResult.getRequestId()));
+        if (inferenceResult.isError() == false) {
+            timingStats.accept(inferenceResult.getTimeMs());
```

Comment:
`LongSummaryStatistics` is not thread safe which is why this method was previously synchronized. 

Perhaps extract the line `timingStats.accept(inferenceResult.getTimeMs());` into a separate synchronized method

Reply:
Ah, well spotted. I added those two lines in a `synchronized` block.